include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -675,80 +1023,397 @@ class charge_distribution_surface<Lyt, false> : public Lyt
+            // charged) is updated.
+            while (charge_quot > 0)
+            {
+                const auto    charge_quot_int = static_cast<int64_t>(charge_quot);
+                const auto    base_int        = static_cast<int64_t>(base);
+                const int64_t quotient_int    = charge_quot_int / base_int;
+                const int64_t remainder_int   = charge_quot_int % base_int;
+                charge_quot                   = static_cast<uint64_t>(quotient_int);
+                // If the current position is not the dependent-cell position, the charge state is updated.
+                if (counter_negative != dependent_cell_index_negative)
+                {
+                    const auto sign = sign_to_charge_state(static_cast<int8_t>(remainder_int - 1));
+                    if (const auto new_chargesign =
+                            this->get_charge_state_by_index(cell_to_index(index_to_two_state_cell(counter_negative)));
+                        new_chargesign != sign)
+                    {
+                        strg->cell_history.emplace_back(
+                            static_cast<uint64_t>(cell_to_index(index_to_two_state_cell(counter_negative))),
+                            charge_state_to_sign(new_chargesign));
+                        this->assign_charge_state_by_cell_index(
+                            cell_to_index(index_to_two_state_cell(counter_negative)), sign, false);
+                    }
+                    counter_negative -= 1;
+                }
+                // If the current position is the dependent cell position, first the counter_negative is decremented by
+                // one to get to the next cell position to update its charge state.
+                else
+                {
+                    counter_negative -= 1;
+                    const auto sign = sign_to_charge_state(static_cast<int8_t>(remainder_int - 1));
+                    if (const auto old_chargesign =
+                            this->get_charge_state_by_index(cell_to_index(index_to_two_state_cell(counter_negative)));
+                        old_chargesign != sign)
+                    {
+                        strg->cell_history.emplace_back(
+                            static_cast<uint64_t>(cell_to_index(index_to_two_state_cell(counter_negative))),
+                            charge_state_to_sign(old_chargesign));
+                        this->assign_charge_state_by_cell_index(
+                            cell_to_index(index_to_two_state_cell(counter_negative)), sign, false);
+                    }
+                    counter_negative -= 1;
+                }
+            }
         }
     }
+    /**
+     * The charge index of the current charge distribution is returned.
+     *
+     * @return A pair with the charge index and the used base.
+     */
+    [[nodiscard]] charge_index_base get_charge_index() const noexcept
+    {
+        return strg->charge_index;
+    }
```

Comment:
The function name implies that it returns the charge index. Yet, it returns a pair of two numbers. Seems like this function does too much. If you require the base as well, make a second function that returns the base.

Reply:
Good point. Thanks.